src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -160,6 +165,8 @@ SGVector<float64_t> CGaussianProcessMachine::get_posterior_variances(
 
 	// get kernel matrix and create eigen representation of it
 	SGMatrix<float64_t> k_tsts=kernel->get_kernel_matrix();
+	//To DO in order to speed up the prediction process,
+	//we only need to obtain the diagonal elements of the kernel matrix
```

Comment:
That's great! Although I'd make sure that adding something generic to the base is actually something that is useful. You can add extra methods in the sub classes (wherever appropriate) and at the time of usage you can just dynamic cast the kernel to call that method.. for things that are not supposed to be general, keeping them limited to the scope where they are actually used helps at times. I don't know much about the context but I hope that it makes sense :)

Great work by the way :) Hope to go through your additions soon!